src/utils/coingecko.ts

```diff
@@ -0,0 +1,100 @@
+import { ChainId } from '@swapr/sdk'
+
+interface PriceInformation {
+  token: string
+  amount: string | null
+}
+
+function getApiUrl(): string {
+  // it's all the same base url
+  return 'https://api.coingecko.com/api'
+}
+
+// Defaults
+const API_NAME = 'Coingecko'
+const API_BASE_URL = getApiUrl()
+const API_VERSION = 'v3'
+const DEFAULT_HEADERS = {
+  'Content-Type': 'application/json'
+}
+
+function _getApiBaseUrl(chainId: ChainId): string {
+  const baseUrl = API_BASE_URL
+
+  if (!baseUrl) {
+    throw new Error(`Unsupported Network. The ${API_NAME} API is not deployed in the Network ${chainId}`)
+  } else {
+    return baseUrl + '/' + API_VERSION
+  }
+}
+
+function _getCoinGeckoAssetPlatform(chainId: ChainId) {
+  switch (chainId) {
+    // Use of asset platforms - supports ethereum and xdai
+    // https://api.coingecko.com/api/v3/asset_platforms
+    case ChainId.MAINNET:
+      return 'ethereum'
+    case ChainId.ARBITRUM_ONE:
+      return 'arbitrum-one'
+    case ChainId.XDAI:
+      return 'xdai'
+    default:
+      return null
+  }
```

Comment:
at some point it would seem more appropriate to make a map of (ChainId, returnValues) and do a get on that map.
...Maybe on a future development

Reply:
Good suggestion, just simplified the code!